crates/sui-sdk/src/apis.rs

```diff
@@ -1133,39 +1136,49 @@ impl QuorumDriverApi {
     ) -> SuiRpcResult<SuiTransactionBlockResponse> {
         let (tx_bytes, signatures) = tx.to_tx_bytes_and_signatures();
         let request_type = request_type.unwrap_or_else(|| options.default_execution_request_type());
-        let mut retry_count = 0;
+
         let start = Instant::now();
-        while retry_count < WAIT_FOR_LOCAL_EXECUTION_RETRY_COUNT {
-            let response: SuiTransactionBlockResponse = self
-                .api
-                .http
-                .execute_transaction_block(
-                    tx_bytes.clone(),
-                    signatures.clone(),
-                    Some(options.clone()),
-                    Some(request_type.clone()),
-                )
-                .await?;
+        let response = self
+            .api
+            .http
+            .execute_transaction_block(
+                tx_bytes.clone(),
+                signatures.clone(),
+                Some(options.clone()),
+                Some(request_type.clone()),
+            )
+            .await?;
 
-            match request_type {
-                ExecuteTransactionRequestType::WaitForEffectsCert => {
-                    return Ok(response);
-                }
-                ExecuteTransactionRequestType::WaitForLocalExecution => {
-                    if let Some(true) = response.confirmed_local_execution {
-                        return Ok(response);
-                    } else {
-                        // If fullnode executed the cert in the network but did not confirm local
-                        // execution, it must have timed out and hence we could retry.
-                        retry_count += 1;
-                    }
+        if let ExecuteTransactionRequestType::WaitForEffectsCert = request_type {
+            return Ok(response);
+        }
+
+        // JSON-RPC ignores WaitForLocalExecution, so simulate it by polling for the transaction.
+        let mut poll_response = tokio::time::timeout(WAIT_FOR_LOCAL_EXECUTION_TIMEOUT, async {
+            // Apply a short delay to give state sync a chance to catch up
```

Comment:
btw I just learned that local execution doesn't always happen through state sync, and in many cases it happens much sooner here: https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/transaction_orchestrator.rs#L437

Reply:
Interesting -- I'll make this comment more general.